src/Service/IndividualKpiValidatorService.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\IndividualKpi;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class IndividualKpiValidatorService
+{
+    private ValidatorInterface $validator;
+    private ValidationMessage $validationMessage;
+
+    public function __construct(ValidatorInterface $validator, ValidationMessage $validationMessage)
+    {
+        $this->validator = $validator;
+        $this->validationMessage = $validationMessage;
+    }
+
+    public function validateIndividualKpi(array $data): array
+    {
+        $individualKpi = new IndividualKpi();
+        $individualKpi->setKpiSetupId($data['kpi_setup_id']);
+        $individualKpi->setKpiTypeId($data['kpi_type_id']);
+        $individualKpi->setPeriodId($data['period_id']);
+        $individualKpi->setCreatedBy(1);
+        $individualKpi->setStatus(1);
+        $errors = $this->validator->validate($individualKpi);
```

Comment:
We are declaring it ($individualKpi = new IndividualKpi();) twice, plz initiate it from previous method & pass it to validator for validation.

Reply:
done with Thanks